src/render/src/macros/structure.rs

```diff
@@ -18,8 +18,20 @@
 macro_rules! gfx_impl_struct {
     ($runtime_format:ty : $compile_format:path = $root:ident {
         $( $field:ident: $ty:ty = $name:expr, )*
+    }) => (gfx_impl_struct_meta! {
+        impl_struct_meta $runtime_format : $compile_format = $root {
+            $( $field : $ty = $name, )*
+        }
+    })
+}
+
+#[macro_export]
+macro_rules! gfx_impl_struct_meta {
```

Comment:
any reason you decided to not just put the extra stuff into `gfx_impl_struct` instead of routing through the new `gfx_impl_struct_meta`?